tests/cases/conformance/es6/arrowFunction/disallowLineTerminatorBeforeArrow.ts

```diff
@@ -0,0 +1,22 @@
+var f1 = ()
+    => { }
+var f2 = (x: string, y: string) /*
+  */  => { }
+var f3 = (x: string, y: number, ...rest)
+    => { }
+var f4 = (x: string, y: number, ...rest) /*
+  */  => { }
+var f5 = (...rest)
+    => { }
+var f6 = (...rest) /*
+  */  => { }
+var f7 = (x: string, y: number, z = 10)
+    => { }
+var f8 = (x: string, y: number, z = 10) /*
+  */  => { }
+
+function foo(func: () => boolean) { }
+foo(()
+    => true);
+foo(()
+    => { return false; });

```

Comment:
Could we get a few more expression-bodied tests?

We should be testing different parsing contexts as well, so kill two birds with one stone:

```TypeScript
module m {
    class City {
        constructor(x: number, thing = ()
            => 100) {
        }
        
        public m = ()
            => 2 * 2 * 2
    }

    export enum Enum {
        claw = (()
            => 10)()
    }

    export var v = x
        => new City(Enum.claw);
}
```